src/main/java/com/cloudbees/jenkins/plugins/bitbucket/server/client/BitbucketServerAPIClient.java

```diff
@@ -337,11 +338,16 @@ public String getRepositoryUri(@NonNull BitbucketRepositoryType type,
         }
 
         AbstractBitbucketEndpoint endpointConfig = BitbucketEndpointConfiguration.get().findEndpoint(baseURL);
-        if (endpointConfig instanceof BitbucketServerEndpoint && ((BitbucketServerEndpoint)endpointConfig).isCallCanMerge()) {
+        if (endpointConfig instanceof BitbucketServerEndpoint) {
             // This is required for Bitbucket Server to update the refs/pull-requests/* references
             // See https://community.atlassian.com/t5/Bitbucket-questions/Change-pull-request-refs-after-Commit-instead-of-after-Approval/qaq-p/194702#M6829
             for (BitbucketServerPullRequest pullRequest : pullRequests) {
-                pullRequest.setCanMerge(getPullRequestCanMergeById(Integer.parseInt(pullRequest.getId())));
+                if (((BitbucketServerEndpoint) endpointConfig).isCallCanMerge()) {
+                    pullRequest.setCanMerge(getPullRequestCanMergeById(Integer.parseInt(pullRequest.getId())));
+                }
+                if (((BitbucketServerEndpoint) endpointConfig).isCallChanges()) {
+                    callPullRequestChangesById(Integer.parseInt(pullRequest.getId()));
+                }
```

Comment:
Better to only cast it once.
Please cast the endpoint once outside the for loop
```java
                BitbucketServerEndpoint endpoint = (BitbucketServerEndpoint) endpointConfig;
```
```suggestion
                if (endpoint.isCallCanMerge()) {
                    pullRequest.setCanMerge(getPullRequestCanMergeById(Integer.parseInt(pullRequest.getId())));
                }
                if (endpoint.isCallChanges()) {
                    callPullRequestChangesById(Integer.parseInt(pullRequest.getId()));
                }
```